lib/matplotlib/dviread.py

```diff
@@ -792,57 +828,61 @@ class PsfontsMap(object):
     file names.
 
     A texmf tree typically includes mapping files called e.g.
-    psfonts.map, pdftex.map, dvipdfm.map. psfonts.map is used by
-    dvips, pdftex.map by pdfTeX, and dvipdfm.map by dvipdfm.
-    psfonts.map might avoid embedding the 35 PostScript fonts (i.e.,
-    have no filename for them, as in the Times-Bold example above),
-    while the pdf-related files perhaps only avoid the "Base 14" pdf
-    fonts. But the user may have configured these files differently.
+    :file:`psfonts.map`, :file:`pdftex.map`, or :file:`dvipdfm.map`.
+    The file :file:`psfonts.map` is used by :program:`dvips`,
+    :file:`pdftex.map` by :program:`pdfTeX`, and :file:`dvipdfm.map`
+    by :program:`dvipdfm`. :file:`psfonts.map` might avoid embedding
+    the 35 PostScript fonts (i.e., have no filename for them, as in
+    the Times-Bold example above), while the pdf-related files perhaps
+    only avoid the "Base 14" pdf fonts. But the user may have
+    configured these files differently.
     """
-    __slots__ = ('_font',)
+    __slots__ = ('_font', '_filename')
 
     def __init__(self, filename):
         self._font = {}
+        self._filename = filename
+        if six.PY3 and isinstance(filename, bytes):
+            self._filename = filename.decode('ascii', errors='replace')
         with open(filename, 'rt') as file:
             self._parse(file)
 
     def __getitem__(self, texname):
+        assert(isinstance(texname, bytes))
         try:
             result = self._font[texname]
         except KeyError:
-            result = self._font[texname.decode('ascii')]
+            matplotlib.verbose.report(textwrap.fill
+                ('A PostScript file for the font whose TeX name is "%s" '
+                 'could not be found in the file "%s". The dviread module '
+                 'can only handle fonts that have an associated PostScript '
+                 'font file. '
+                 'This problem can often be solved by installing '
+                 'a suitable PostScript font package in your (TeX) '
+                 'package manager.' % (texname.decode('ascii'),
+                                       self._filename),
+                 break_on_hyphens=False, break_long_words=False),
+                'helpful')
```

Comment:
Especially because this line is in a different function call than the ones above it.